src/app/modules/item/components/item-header/item-header.component.ts

```diff
@@ -1,11 +1,19 @@
-import { Component, Input } from '@angular/core';
-import { Item } from '../../http-services/get-item-by-id.service';
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { ItemData } from '../../services/item-datasource.service';
 
 @Component({
   selector: 'alg-item-header',
   templateUrl: './item-header.component.html',
   styleUrls: ['./item-header.component.scss']
 })
-export class ItemHeaderComponent {
-  @Input() item: Item;
+export class ItemHeaderComponent implements OnChanges {
+  @Input() itemData: ItemData;
+
+  current_score = 0;
+
+  ngOnChanges(_changes: SimpleChanges): void {
+    if (this.itemData.results && this.itemData.currentResult) {
+      this.current_score = Math.floor(this.itemData.currentResult.score);
```

Comment:
Probably it would be more useful to do this rounding in the score ring component directly.
And cleaner/easier way to do it is to just use a pipe : https://angular.io/api/common/DecimalPipe

Reply:
fixed

Reply:
Good but you can remove this code then.